pkg/genericapiserver/options/etcd.go

```diff
@@ -45,13 +45,12 @@ func NewEtcdOptions() *EtcdOptions {
 	}
 }
 
-func (s *EtcdOptions) Validate() []error {
-	allErrors := []error{}
+func (s *EtcdOptions) Validate() error {
```

Comment:
API object validation return a list and I think that pattern works well for validation.  I think we should keep this as a slice.